Mage/src/main/java/mage/abilities/keyword/SpreeAbility.java

```diff
@@ -0,0 +1,28 @@
+package mage.abilities.keyword;
+
+import mage.abilities.StaticAbility;
+import mage.cards.Card;
+import mage.constants.Zone;
+
+/**
+ * @author TheElk801
+ */
+public class SpreeAbility extends StaticAbility {
+
+    public SpreeAbility(Card card) {
+        super(Zone.ALL, null);
+        this.setRuleVisible(false);
+        card.getSpellAbility().getModes().setChooseText("Spree <i>(Choose one or more additional costs.)</i>");
+        card.getSpellAbility().getModes().setMinModes(1);
```

Comment:
So the spell should not be considered castable if none of the modes can be paid for.

Not sure whether the test framework supports this - have you tested it?

Reply:
oh I didn't even consider that, I'll look into it

Reply:
Looking into it, I'm not sure how to actually handle that aspect. I know that it won't be able to successfully cast unless it's actually able to pay but I don't know how to prevent it from being castable in the first place.

Reply:
Its possible to implement in getAvailableModes method:

![IMG_0103](https://github.com/magefree/mage/assets/8344157/6b25c2b0-9855-4aa1-9952-e70d2b1dfcf2)

Current method do not check targets or cost, but it can. Like AI code do with targets:
![IMG_0104](https://github.com/magefree/mage/assets/8344157/abb6f799-e5c3-42e6-8afd-ba64f32c327d)

I don’t think it’s a part of that PR. Maybe some of the game engine improves.